packages/ruleset/src/functions/required-array-properties-in-response.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2017 - 2024 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const {
+  isArraySchema,
+  isObject,
+  validateSubschemas,
+} = require('@ibm-cloud/openapi-ruleset-utilities');
+const { LoggerFactory } = require('../utils');
+
+let ruleId;
+let logger;
+
+module.exports = function (schema, _opts, context) {
+  if (!logger) {
+    ruleId = context.rule.name;
+    logger = LoggerFactory.getInstance().getLogger(ruleId);
+  }
+
+  return validateSubschemas(
+    schema,
+    context.path,
+    checkForOptionalArrays,
+    true,
+    false
+  );
+};
+
+/**
+ * Checks "schema" for any optional array properties.
+ * @param {*} schema the schema to check
+ * @param {*} path the json path location of "schema"
+ * @returns
+ */
+function checkForOptionalArrays(schema, path) {
+  const errors = [];
+
+  // If "schema" defines properties, then add a warning for each optional array property.
```

Comment:
I know this is picky too, just a suggestion for precision

```suggestion
  // If "schema" defines properties, then add an error for each optional array property.
```

Reply:
fixed